src/main/java/org/opensearch/ad/util/Bwc.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.ad.util;
+
+import org.opensearch.Version;
+
+/**
+ * A helper class for various feature backward compatibility test
+ *
+ */
+public class Bwc {
+    /**
+     * We are gonna start supporting multi-category field since version 1.1.0.
+     * Since Opendistro and Elasticsearch will use a version after 1.0.0, we
+     * have to check Lucene version as well. Elasticsearch 7.10 uses Lucene 8.7.0
+     *
+     * @param version test version
+     * @return whether the version support multiple category fields
+     */
+    public static boolean supportMultiCategoryFields(Version version) {
+        return version.after(Version.V_1_0_0) && version.luceneVersion.onOrAfter(org.apache.lucene.util.Version.LUCENE_8_8_2);
```

Comment:
Can you specify in the comments that OpenSearch 1.1.0 will be based on Lucene 8.8.2?

Reply:
yeah, added.

Reply:
This looks like common problem when we do version upgrade. I think OpenSearch core team should consider the case that "Opendistro and Elasticsearch will use a version after 1.0.0" in `version.after(Version v)` method, so we don't need to worry about this and check lucene version in all places/plugins.

From https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/Version.java#L222, the version comparison in `after` method is based on `id`, not sure if OpenSearch core team can make sure the `id` of OpenSearch version always greater than ES versions. Can you confirm with @nknize or @adnapibar as I can see they edited the Version class.

Reply:
I checked the code, the id of OpenSearch and Elasticsearch has no such after guarantee you want. 

For example, OpenSearcdh V_1_1_0's id is 1010099, while ElasticSearch's V_7_10_2's id is 7100299.  If we just compare id, ElasticSearch is after V_1_1_0.  So it is wrong. I can confirm with the people you mentioned.

Reply:
The version comparison works (i.e OpenSearch 1.x is after Elasticsearch v7.10.2) due to this logic https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/Version.java#L210-L211

Reply:
Thanks for confirming, Rabi.  I will remove the check for lucene version.